runners/local-artifact-service-java/src/test/java/org/apache/beam/artifact/local/LocalFileSystemArtifactStagerServiceTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.artifact.local;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import com.google.protobuf.ByteString;
+import io.grpc.stub.StreamObserver;
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.Artifact;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.ArtifactChunk;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.CommitManifestRequest;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.CommitManifestResponse;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.Manifest;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.PutArtifactRequest;
+import org.apache.beam.sdk.common.runner.v1.ArtifactApi.PutArtifactResponse;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link LocalFileSystemArtifactStagerService}. */
+@RunWith(JUnit4.class)
+public class LocalFileSystemArtifactStagerServiceTest {
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private LocalFileSystemArtifactStagerService stager;
+  private File stagingLocation;
+
+  @Before
+  public void setup() throws Exception {
+    stagingLocation = temporaryFolder.newFolder();
+    stager = LocalFileSystemArtifactStagerService.withRootDirectory(stagingLocation);
+  }
+
+  @Test
+  public void singleDataPutArtifactSucceeds() throws Exception {
+    byte[] data = "foo-bar-baz".getBytes();
+    RecordingStreamObserver<PutArtifactResponse> responseObserver = new RecordingStreamObserver<>();
+    StreamObserver<PutArtifactRequest> requestObserver = stager.putArtifact(responseObserver);
+
+    String name = "my-artifact";
+    requestObserver.onNext(PutArtifactRequest.newBuilder().setName(name).build());
```

Comment:
Incidentally, it is probably best practice to make this `oneof content` so that each of the alternatives is a full message. That way the `string name` alternative is actually `InitialArtifactMetadata` or some such and can grow or change in the future.

Reply:
Done.